src/nb_type.cpp

```diff
@@ -484,7 +486,7 @@ PyObject *nb_type_new(const type_data *t) noexcept {
 
     const char *tp_name = PyUnicode_AsUTF8AndSize(temp_ht->ht_name, nullptr);
 
-    PyObject *result = PyType_GenericAlloc(metaclass, 0);
+    PyObject *result = PyType_GenericAlloc(metaclass, num_members ? num_members + 1 : 0);
```

Comment:
I feel like this is missing some kind of adaptation to the actual number of members in `temp_tp`, which was created by `PyType_FromSpec`. What if that process adds further members? Copying `num_members + 1` elements then would not produce a copy with a null-initialized last (sentinel) element.

Reply:
I realized there's a better way to do this, which is to copy `Py_SIZE(x) * tp_itemsize` bytes.